File: Logitude/AngularModules/AngularModules/Shipments/EntityPMs/DocumentTypePM.ts

import {UIProperties, UIProperty} from '../../Infrastructure/Components/LogitudeComponents/UIProperties';
export class DocumentTypePM {


    public UIProperties: UIProperties;

    constructor() {
        this.UIProperties = new UIProperties;
        this.IsDirty = false;
    }

    public IsDirty: boolean;


    private _Id: string;
    public get Id() { return this._Id; }
    public set Id(newValue: string) { this._Id = newValue; this.MarkAsDirty(); }



    private _Tenant: number;
    public get Tenant() { return this._Tenant; }
    public set Tenant(newValue: number) { this._Tenant = newValue; this.MarkAsDirty(); }

    private _Code: string;
    public get Code() { return this._Code; }
    public set Code(newValue: string) { this._Code = newValue; this.MarkAsDirty(); }

    private _Name: string;
    public get Name() { return this._Name; }
    public set Name(newValue: string) { this._Name = newValue; this.MarkAsDirty(); }

    private _Notes: string;
    public get Notes() { return this._Notes; }
    public set Notes(newValue: string) { this._Notes = newValue; this.MarkAsDirty(); }



    private _IsAir: boolean;
    public get IsAir() { return this._IsAir; }
    public set IsAir(newValue: boolean) { this._IsAir = newValue; this.MarkAsDirty(); }



    private _IsOcean: boolean;
    public get IsOcean() { return this._IsOcean; }
    public set IsOcean(newValue: boolean) { this._IsOcean = newValue; this.MarkAsDirty(); }


    private _IsInland: boolean;
    public get IsInland() { return this._IsInland; }
    public set IsInland(newValue: boolean) { this._IsInland = newValue; this.MarkAsDirty(); }


    private _IsDocIn: boolean;
    public get IsDocIn() { return this._IsDocIn; }
    public set IsDocIn(newValue: boolean) { this._IsDocIn = newValue; this.MarkAsDirty(); }



    private _IsDocOut: boolean;
    public get IsDocOut() { return this._IsDocOut; }
    public set IsDocOut(newValue: boolean) { this._IsDocOut = newValue; this.MarkAsDirty(); }



    private _InActive: boolean;
    public get InActive() { return this._InActive; }
    public set InActive(newValue: boolean) { this._InActive = newValue; this.MarkAsDirty(); }

    private _TemplateFormatCode: string;
    public get TemplateFormatCode() { return this._TemplateFormatCode; }
    public set TemplateFormatCode(newValue: string) { this._TemplateFormatCode = newValue; this.MarkAsDirty(); }


    private _IsMaster: boolean;
    public get IsMaster() { return this._IsMaster; }
    public set IsMaster(newValue: boolean) { this._IsMaster = newValue; this.MarkAsDirty(); }




    private _IsDirect: boolean;
    public get IsDirect() { return this._IsDirect; }
    public set IsDirect(newValue: boolean) { this._IsDirect = newValue; this.MarkAsDirty(); }



    private _IsHouse: boolean;
    public get IsHouse() { return this._IsHouse; }
    public set IsHouse(newValue: boolean) { this._IsHouse = newValue; this.MarkAsDirty(); }


    private _IsReadOnly: boolean;
    public get IsReadOnly() { return this._IsReadOnly; }
    public set IsReadOnly(newValue: boolean) { this._IsReadOnly = newValue; this.MarkAsDirty(); }

    private _Subject: string;
    public get Subject() { return this._Subject; }
    public set Subject(newValue: string) { this._Subject = newValue; this.MarkAsDirty(); }

    private _FollowUpTypeId: string;
    public get FollowUpTypeId() { return this._FollowUpTypeId; }
    public set FollowUpTypeId(newValue: string) { this._FollowUpTypeId = newValue; this.MarkAsDirty(); }



    private _FollowUpTypeName: string;
    public get FollowUpTypeName() { return this._FollowUpTypeName; }
    public set FollowUpTypeName(newValue: string) { this._FollowUpTypeName = newValue; this.MarkAsDirty(); }



    private _ObjectTableId: string;
    public get ObjectTableId() { return this._ObjectTableId; }
    public set ObjectTableId(newValue: string) { this._ObjectTableId = newValue; this.MarkAsDirty(); }


    private _ChildEntityId: string;
    public get ChildEntityId() { return this._ChildEntityId; }
    public set ChildEntityId(newValue: string) { this._ChildEntityId = newValue; this.MarkAsDirty(); }

    private _ChildEntityReference: string;
    public get ChildEntityReference() { return this._ChildEntityReference; }
    public set ChildEntityReference(newValue: string) { this._ChildEntityReference = newValue; this.MarkAsDirty(); }


MarkAsDirty() {
    this.IsDirty = true;
}
}

File: Logitude/Logitude.BL/CommonDataModel/EntityPMs/DocumentTypePM.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Simplog.Server.Infrastructure;
using System.ComponentModel.DataAnnotations;
using System.ServiceModel.DomainServices.Server;
using Logitude.BL.Validators;

namespace Logitude.BL.CommonDataModel.EntityPMs
{
    [CustomValidation(typeof(DocumentTypeClassLevelValidator),"ValidateClass")]
    [CustomValidation(typeof(Validators.ClassLevelValidator), "ValidateClass")]
    [CustomValidation(typeof(DocumentTypeTransModeValidator), "IsDocumentTypeValid")]
    [DataContract]
    public class DocumentTypePM
    {
        [Key]
        [DataMember]
        public string Id { get; set; }
        [DataMember]
        public int Tenant { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public string Code { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public string Name { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public string Notes { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsAir { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsOcean { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsInland { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsDocIn { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsDocOut { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool InActive { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public string TemplateFormatCode { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsMaster { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsDirect { get; set; }

        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        [DataMember]
        public bool IsHouse { get; set; }

        [StringLength(500, MinimumLength = 0, ErrorMessage = "The length of the subject must be less than 500 character!")]
        [DataMember]
        public string Subject { get; set; }
        [DataMember]
        public string FollowUpTypeId { get; set; }
        [DataMember]
        public string FollowUpTypeName { get; set; }
        [DataMember]
        public string ObjectTableId { get; set; }
        [DataMember]
        public string SearchFields { get; set; }
        [DataMember]
        public string CustomControl { get; set; }
        [DataMember]
        public string CustomerRoleId { get; set; }
        [DataMember]
        public string AgentRoleId { get; set; }
        [DataMember]
        public string DocumentTypeDefaultReportTemplateId { get; set; }
        [DataMember]
        public string DocumentTypeDefaultHTMLTemplateId { get; set; }
        [DataMember]
        public string DocumentTypeDefaultEditorTool { get; set; }
        [DataMember]
        public bool IsAgentView { get; set; }
        [DataMember]
        public bool IsCustomerView { get; set; }
        [DataMember]
        public bool IsHybrid { get; set; }

        [DataMember]
        public string ObjectTableName { get; set; }

        [DataMember]
        public bool IsReadOnly { get; set; }

          [DataMember]
        public bool IsEnabledForCustomers { get; set; }
           [DataMember]       
        public bool IsCopiedAtSignup { get; set; }
          [DataMember]
        public string CountryCode { get; set; }


        [DataMember]
        public bool IsDocumentOneTimePrintLimited { get; set; }
        [DataMember]
        public string LimitedPrintCopyId { get; set; }

        [DataMember]
        public bool IsAgentSharedInMaster { get; set; }
        [DataMember]
        public bool IsAgentSharedInDirect { get; set; }
        [DataMember]
        public bool IsAgentSharedInHouse { get; set; }
        [DataMember]
        public string SharedDocumentTypeCopyId { get; set; }
        [DataMember]
        public bool IsAirDigitalSignRequired { get; set; }
        [DataMember]
        public bool IsOceanDigitalSignRequired { get; set; }
        [DataMember]
        public bool IsInlandDigitalSignRequired { get; set; }


        private List<DocumentTypeCustomFieldPM> documentTypeCustomFields;
        [Include]
        [Association("DocumentTypePMDocumentTypeCustomFieldPM", "Id", "DocumentTypeId")]
        [DataMember]
        public List<DocumentTypeCustomFieldPM> DocumentTypeCustomFields
        {
            get 
            {
                if (documentTypeCustomFields == null)
                {
                    documentTypeCustomFields = new List<DocumentTypeCustomFieldPM>();
                }
                return documentTypeCustomFields; 
            }
            set { documentTypeCustomFields = value; }
        }

        private List<DocumentTypeCopyPM> documentTypeCopies;
        [Composition]
        [Include]
        [Association("DocumentTypePMDocumentTypeCopyPM", "Id", "DocumentTypeId")]
        [DataMember]
        public virtual List<DocumentTypeCopyPM> DocumentTypeCopies
        {
            get
            {
                if (documentTypeCopies == null)
                {
                    documentTypeCopies = new List<DocumentTypeCopyPM>();
                }
                return documentTypeCopies;
            }
            set { documentTypeCopies = value; }
        }
        
        public List<DocumentTypeTemplatePM> DocumentTypeTemplates
        {
            get
            ;
            set;
        }

        [DataMember]
        [CustomValidation(typeof(Validators.ValidationClass), "ValidateClass")]
        public string DocumentTypeCategoryCode { get; set; }

        [DataMember]
        public string DocumentTypeCategoryName { get; set; }
        [DataMember]
        public int OrderBy { get; set; }

        [DataMember]
        public string FileName { get; set; }


        [DataMember]
        public bool IsSystemAdditionalPrintingFields { get; set; }
        [DataMember]
        public string PrintingFieldsScreenCode { get; set; }
    }
}
